cellprofiler/modules/imagegradient.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+"""
+
+<strong>Image gradient</strong>
+
+Compute the local gradient of the image. The image gradient is useful for finding boundaries of objects. In a gradient
+image, pixels at the edges of bright regions of interest have the brightest intensities. Pixels in the background or
+in the centers of regions of interest have zero or dimmer intensity.
+
+"""
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+import numpy
+import skimage.filters
+import skimage.morphology
+
+
+class ImageGradient(cellprofiler.module.ImageProcessing):
+    module_name = "ImageGradient"
+
+    variable_revision_number = 1
+
+    def create_settings(self):
+        super(ImageGradient, self).create_settings()
+
+        self.structuring_element = cellprofiler.setting.StructuringElement(
+            doc="""Neighborhood in which to compute the local gradient. Select a two-dimensional shape such as "disk"
+            for images, and a three-dimensional shape such as "ball" for volumes. A larger size will compute the gradient
+            over larger patches of the image and can obscure smaller features."""
+        )
+
+    def settings(self):
+        __settings__ = super(ImageGradient, self).settings()
+
+        return __settings__ + [
+            self.structuring_element
+        ]
+
+    def visible_settings(self):
+        __settings__ = super(ImageGradient, self).visible_settings()
+
+        return __settings__ + [
+            self.structuring_element
+        ]
+
+    def run(self, workspace):
+        x_name = self.x_name.value
+
+        y_name = self.y_name.value
+
+        images = workspace.image_set
+
+        x = images.get_image(x_name)
+
+        x_data = x.pixel_data
+
+        if x.dimensions is 3 or x.multichannel:
+            y_data = numpy.zeros_like(x_data)
+
+            for z, image in enumerate(x_data):
+                y_data[z] = skimage.filters.rank.gradient(image, self.__structuring_element())
+        else:
+            y_data = skimage.filters.rank.gradient(x_data, self.structuring_element.value)
```

Comment:
There is no non-local gradient filter in scikit-image: http://scikit-image.org/docs/dev/api/skimage.filters.html